src/unix/spawn-helper.cc

```diff
@@ -0,0 +1,63 @@
+#include <fcntl.h>
+#include <signal.h>
+#include <string>
+#include <sys/ioctl.h>
+#include <sys/resource.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include "comms.h"
+
+int main (int argc, char** argv) {
+  sigset_t empty_set;
+  sigemptyset(&empty_set);
+  pthread_sigmask(SIG_SETMASK, &empty_set, nullptr);
+
+  struct rlimit rlim_ofile;
+  getrlimit(RLIMIT_NOFILE, &rlim_ofile);
+  for (rlim_t fd = STDERR_FILENO + 1; fd < rlim_ofile.rlim_cur; fd++) {
+    if (fd != COMM_PIPE_FD) {
+      close(fd);
+    }
+  }
```

Comment:
Nice, should solve all leaking fd problems, even without that CLOEXEC flag in spawn :smile: